pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -168,6 +177,10 @@ func (f *ring0Factory) DiscoveryClient() (discovery.CachedDiscoveryInterface, er
 	return f.discoveryFactory.DiscoveryClient()
 }
 
+func (f *ring0Factory) BindDiscoveryFlags(flags *pflag.FlagSet) {
```

Comment:
would prefer not to expose this... can we just make ClientAccessFactory expose `DiscoveryClient() (discovery.CachedDiscoveryInterface, error)`?

Reply:
I wanted to do that, but the `ClientAccessFactory` is embedded in the `Factory`:

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/util/factory.go#L76-L80

Reply:
Does it have to be embedded or can we just share the one method between the interfaces?

Reply:
Re-reading your comment, yeah, I'll try that.